src/less/dialog/ds6/dialog-base.less

```diff
@@ -0,0 +1,201 @@
+    position: absolute;
+    margin: auto;
+    top: 15%;
+    left: 0;
+    right: 0;
+    margin-bottom: 15vh;
+    background-color: @ds6-color-g201-gray;
+    will-change: opacity, transform;
+    .dialog-grid(@cols: 16, @gutter: 16px, @min: 8, @max: 14);
+}
+
+.dialog__mask {
+    position: fixed;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    opacity: 0.7;
+    background-color: @ds6-color-text-default;
+    will-change: opacity;
+}
+
+.dialog__body {
+    padding: 16px;
+    width: 100%;
+    min-height: 100%;
+    box-sizing: border-box;
+
+    & :first-child {
+        margin-top: 0;
+        padding-top: 0;
+    }
+
+    & :last-child {
+        margin-bottom: 0;
+        padding-bottom: 0;
+    }
+}
+
+.dialog__close {
+    position: absolute;
+    top: 0;
+    right: 0;
+    padding: 16px;
+    border: 0;
+    cursor: pointer;
+    color: @ds6-color-text-default;
+    background: @ds6-color-g201-gray;
+
+    &:hover,
+    &:focus {
+        color: @ds6-color-link-default;
+    }
+}
+
+// MODIFIERS
+//-----------------------------
+
+.dialog__window--full,
+.dialog__window--left,
+.dialog__window--right {
+    top: 0;
+    height: 100%;
+    margin-bottom: 0;
+    overflow-y: auto;
+    -webkit-overflow-scrolling: touch;
+
+    .dialog__close {
+        position: -webkit-sticky;
+        position: sticky;
+        float: right;
+    }
+}
+
+.dialog__window--fill {
+    min-height: 70%;
+}
+
+.dialog__window--full {
+    max-width: 100%;
+    width: 100%;
+    bottom: 0;
+}
+
+.dialog__window--left {
+    right: auto;
+    border-right: 1px solid rgba(153, 153, 153, 0.18);
+}
+
+.dialog__window--right {
+    left: auto;
+    border-left: 1px solid rgba(153, 153, 153, 0.18);
+}
+
+// TRANSITIONS
+//-----------------------------
+
+// Prime transitions.
+.dialog-show:not([hidden]),
+.dialog-hide[hidden] {
+    .dialog__window--fade,
+    .dialog__mask--fade {
+        transition: opacity 0.16s ease-out;
+    }
+
+    .dialog__window--slide {
+        transition: transform 0.32s ease-out;
+    }
+
+    .dialog__mask--fade-slow {
+        transition: opacity 0.32s ease-out;
+    }
+}
+
+// Apply hide animation.
+.dialog-hide,
+.dialog-show--from {
+    display: block !important;
+
+    .dialog__window--fade,
+    .dialog__mask--fade,
+    .dialog__mask--fade-slow {
+        opacity: 0;
+    }
+
+    .dialog__window--slide {
+        &.dialog__window--left {
+            transform: translateX(-100%);
+        }
+
+        &.dialog__window--right {
+            transform: translateX(100%);
+        }
+    }
+}
+
+// Apply show animation.
+.dialog-show,
+.dialog-hide--from {
+    display: block !important;
+
+    .dialog__window--fade {
+        opacity: 1;
+    }
+
+    .dialog__window--slide {
+        transform: translateX(0);
+    }
+
+    .dialog__mask--fade,
+    .dialog__mask--fade-slow {
+        opacity: 0.7;
+    }
+}
+
+// MEDIA QUERIES
+//-----------------------------
+
+@media(max-width: 600px) {
```

Comment:
As with other comments elsewhere, we should consider being mobile first. Hence, making the media query for `min-width: 601px` and adjusting the LESS mixins accordingly (e.g. flip the mixins below with the current mixins for the `.dialog_window`).

Thoughts?

Reply:
Looks like this would change some of the behavior of the `.dialog__window--full`.

Reply:
I have no issues with that. I'll update the PR shortly.

Reply:
@DylanPiercey You may have picked up on this already, but FYI we don't use media queries for 600px breakpoint. We use an adaptive approach instead. Any styles in `dialog-base.less` are always considered our base styles. If we need to make adjustments for over 600px, then those rules go in a `dialog-large.less` file and are included adaptively by Lasso.

Reply:
We currently don't have a `skin-large.less` for DS6, or even a `skin-full.less`. Are you suggesting they need to be added, too?

Reply:
We currently don't have a `skin-large.less` for DS6, or even a `skin-full.less`. Are you suggesting they need to be added?